server/sonar-server/src/main/java/org/sonar/server/ce/ws/TaskFormatter.java

```diff
@@ -133,27 +143,37 @@ public TaskFormatter(DbClient dbClient, System2 system2) {
     return builder.build();
   }
 
-  private static WsCe.Task.Builder buildComponent(WsCe.Task.Builder builder, @Nullable ComponentDto componentDto) {
-    if (componentDto != null) {
-      builder.setComponentKey(componentDto.getDbKey());
-      builder.setComponentName(componentDto.name());
-      builder.setComponentQualifier(componentDto.qualifier());
+  private static WsCe.Task.Builder buildComponent(WsCe.Task.Builder builder, @Nullable String componentUuid, DtoCache componentDtoCache) {
+    ComponentDto componentDto = componentDtoCache.getComponent(componentUuid);
+    if (componentDto == null) {
+      return builder;
     }
+    builder.setComponentKey(componentDto.getKey());
+    builder.setComponentName(componentDto.name());
+    builder.setComponentQualifier(componentDto.qualifier());
+    String branch = componentDto.getBranch();
+    if (branch == null) {
+      return builder;
+    }
+    builder.setBranch(branch);
+    builder.setBranchType(Common.BranchType.valueOf(componentDtoCache.getBranch(componentUuid).getBranchType().name()));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "getBranch" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2259)